### ISSUE TITLE
`KubeMemOvercommit` alert wrong?

### ISSUE THREAD

**Issue by eyalzek** [kubernetes-monitoring/kubernetes-mixin#162]
As discussed in: https://github.com/coreos/prometheus-operator/issues/2406

`KubeMemOvercommit` started firing recently so I took a look at the expression defined here: https://github.com/kubernetes-monitoring/kubernetes-mixin/blob/2728a81295761a9031b8152b37c2efb05ca949aa/alerts/resource_alerts.libsonnet#L24-L42

```
sum(namespace_name:kube_pod_container_resource_requests_memory_bytes:sum)
          /
        sum(node_memory_MemTotal_bytes)
          >
        (count(node:node_num_cpu:sum)-1)
          /
        count(node:node_num_cpu:sum)
```

it's checking how much memory is used on all nodes from what's available and then compares this value to the total CPUs -1 / total CPUs ? Why is a comparison to the CPUs a factor in alerting of over committing memory? Am I reading this wrong/missing something?

**Comment by metalmatze** [comment]
Looking at the history of the file, @tomwilkie should know more about this query and if it is actually broken.

**Comment by cbeneke** [comment]
The alert should be fine: `count(node:node_num_cpu:sum)` is the number of metrics in `node:node_num_cpu:sum` ignoring the actual value. As each node has exactly one metric for the CPU count, this is the number of nodes.
Thus the comparison is agains (#nodes -  1) / (#nodes), which means you can tolerate one node failure without breaching your (then) existing memory limitations.

**Comment by varac** [comment]
> count(node:node_num_cpu:sum) is the number of metrics in node:node_num_cpu:sum ignoring the actual value. As each node has exactly one metric for the CPU count, this is the number of nodes

Ok, tecnically this makes sense but isn't there a simpler expression/metric to get the numer of nodes ?

**Comment by paulfantom** [comment]
I am closing the issue as it should be fixed with #646. If you still see the problem, please reopen.
